src/unix/unix_c/unix_get_pw_gr_nam_id_job.c

```diff
@@ -81,9 +82,17 @@ static value alloc_group_entry(struct group *entry)
     {                                                                  \
         size_t buffer_size = sysconf(_SC_##CONF##_R_SIZE_MAX);         \
         if (buffer_size == (size_t)-1) buffer_size = 16384;            \
-        job->buffer = (char *)lwt_unix_malloc(buffer_size);            \
-        job->result = FUNC##_r(job->ARG, &job->entry, job->buffer,     \
-                               buffer_size, &job->ptr);                \
+        while (1)                                                   \
+        {                                                              \
+            job->buffer = (char *)lwt_unix_malloc(buffer_size);        \
+            job->result = FUNC##_r(job->ARG, &job->entry, job->buffer, \
+                                   buffer_size, &job->ptr);            \
+            if (job->result == 0) break;                               \
+            if (job->result != ERANGE) break;                          \
+            free(job->buffer);                                         \
+            buffer_size = buffer_size * 2;                             \
+            if (buffer_size > 1 << 30) break;                          \
```

Comment:
> I'm checking if it's bigger than 2^30. I picked that constant sort of randomly: I wanted something less than 2^32 for compatibility with 32-bit architectures, but otherwise not real reason.

That actually wouldn't be compatible with 32-bit architectures if it ever reached anywhere near that limit, since most 32-bit OSs probably wouldn't let you allocate a 1GB buffer. I don't think we would ever keep failing with `ERANGE` until reaching such a size, but I think, for defensive programming purposes, it's best to choose something smaller (especially since we are building up to this in powers of 2, and in case of some kind of heap fragmentation, we are risking allocating almost 2GB of memory in a geometric progression, say, on a buggy system). I suggest picking 1MB (`1 << 20`). This should be almost risk-free on most systems, and it's plenty enough for the 16K that seemed to solve the original issue.

Reply:
Fixed.